python/jmxquery/__init__.py

```diff
@@ -50,125 +60,156 @@ def __init__(self,
-
         :return:    The query string to find the MBean in format:
-
                         {mBeanName}/{attribute}/{attributeKey}
-
                     Example: java.lang:type=Memory/HeapMemoryUsage/init
         """
-        query = ""
+        query = []
         if self.metric_name:
-            query += self.metric_name
-
-            if ((self.metric_labels != None) and (len(self.metric_labels) > 0)):
-                query += "<"
-                keyCount = 0
-                for key, value in self.metric_labels.items():
-                    query += key + "=" + value
-                    keyCount += 1
-                    if keyCount < len(self.metric_labels):
-                        query += ","
-                query += ">"
-            query += "=="
-
-        query += self.mBeanName
+            query.append(self.metric_name)
+
+            if self.metric_labels != None:
+                if len(self.metric_labels) > 0:
+                    query.append("<")
+                    keyCount = 0
+                    for key, value in self.metric_labels.items():
+                        query.append("%s = %s" % (
+                            key,
+                            value
+                        ))
+                        keyCount += 1
+                        if keyCount < len(self.metric_labels):
+                            query.append(",")
+                    query.append(">")
+            query.append("==")
+
+        query.append(self.mBeanName)
         if self.attribute:
-            query += "/" + self.attribute
+            query.append("/%s" % self.attribute)
         if self.attributeKey:
-            query += "/" + self.attributeKey
+            query.append("/%s" % self.attributeKey)
 
-        return query
+        return ''.join(query)
 
     def to_string(self):
 
-        string = ""
+        string = []
         if self.metric_name:
-            string += self.metric_name
-
-            if ((self.metric_labels != None) and (len(self.metric_labels) > 0)):
-                string += " {"
-                keyCount = 0
-                for key, value in self.metric_labels.items():
-                    string += key + "=" + value
-                    keyCount += 1
-                    if keyCount < len(self.metric_labels):
-                        string += ","
-                string += "}"
+            string.append(self.metric_name)
+
+            if self.metric_labels != None:
+                if len(self.metric_labels) > 0:
+                    string.append(" {")
+                    keyCount = 0
+                    for key, value in self.metric_labels.items():
+                        string.append("%s = %s" % (
+                            key,
+                            value
+                        ))
+                        keyCount += 1
+                        if keyCount < len(self.metric_labels):
+                            string.append(",")
```

Comment:
This loop can be written in the same way I suggested the loop above to be re-written.

Reply:
Implemented per your recommendation.